starport/pkg/spn/chain.go

```diff
@@ -2,13 +2,129 @@ package spn
 
 import (
 	"context"
-	"github.com/tendermint/starport/starport/pkg/jsondoc"
+	"golang.org/x/sync/errgroup"
 	"time"
 
 	"github.com/cosmos/cosmos-sdk/types"
 	genesistypes "github.com/tendermint/spn/x/genesis/types"
+	"github.com/tendermint/starport/starport/pkg/jsondoc"
 )
 
+// ChainSummary represents the summary of a chain in Genesis module of SPN.
+type ChainSummary struct {
+	ChainID            string
+	Source             string
+	TotalValidators    int
+	ApprovedValidators int
+	TotalProposals     int
+	ApprovedProposals  int
+}
+
+// ChainList lists chain summaries
+func (c *Client) ChainList(ctx context.Context, accountName string) ([]ChainSummary, error) {
+	clientCtx, err := c.buildClientCtx(accountName)
+	if err != nil {
+		return []ChainSummary{}, err
+	}
+
+	// List the chains
+	q := genesistypes.NewQueryClient(clientCtx)
+	req := &genesistypes.QueryListChainsRequest{}
+	chainList, err := q.ListChains(ctx, req)
+	if err != nil {
+		return []ChainSummary{}, err
+	}
+
+	// Get the summary of each chain
+	chainSummaries := make([]ChainSummary, len(chainList.Chains))
+	chainSummariesGroup := new(errgroup.Group)
```

Comment:
```suggestion
	g, ctx := errgroup.WithCancel(ctx)
```
Better if we use above to get derived ctx to be able to cancel all SPN requests if one of these requests goes wrong.